core/eolearn/core/eodata.py

```diff
@@ -162,6 +191,25 @@ def _parse_feature_value(self, value: object, feature_name: str) -> np.ndarray:
 
         return value
 
+    def _update_temporal_dim(self, new_value: int | None) -> None:
+        self._temporal_dim = new_value
+        if self._temporal_dim is None:
+            if self._content:
```

Comment:
Here you are checking if the features exist, but I'm missing the point where you check if this is a FeatureDictNumpy of temporal features or not.

In other words, I would expect this to be `if self._content and self.feature_type.is_temporal()`

Reply:
i didn't do this because temporal dimensions are only updated for temporal feature dictionaries. I guess it wouldn't hurt to have another check here as well.